haystack/utils/import_utils.py

```diff
@@ -62,14 +60,27 @@ def _optional_component_not_installed(component: str, dep_group: str, source_err
     ) from source_error
 
 
-def fetch_archive_from_http(url: str, output_dir: str, proxies: Optional[dict] = None) -> bool:
+def fetch_archive_from_http(
+    url: str,
+    output_dir: str,
+    proxies: Optional[Dict[str, str]] = None,
+    timeout: Union[float, Tuple[float, float]] = 10.0,
+) -> bool:
     """
     Fetch an archive (zip, gz or tar.gz) from a url via http and extract content to an output directory.
 
     :param url: http address
+    :type url: str
```

Comment:
No need to add the types here as we already have them in the type annotations in the methods signature